examples/react/ssr/tanstack-router-app/app.config.js

```diff
@@ -1,100 +1,65 @@
+import { TanStackRouterVite } from "@tanstack/router-vite-plugin";
+import { serverFunctions } from "@vinxi/server-functions/plugin";
+import { serverTransform } from "@vinxi/server-functions/server";
 import reactRefresh from "@vitejs/plugin-react";
-import { join } from "path";
-import { createApp, resolve } from "vinxi";
-import {
-	BaseFileSystemRouter,
-	analyzeModule,
-	cleanPath,
-} from "vinxi/fs-router";
+import { resolve } from "import-meta-resolve";
+import * as path from "path";
+import { createApp } from "vinxi";
 
-function tanstackFileRouter(config) {
-	class TanstackFileRouter extends BaseFileSystemRouter {
-		glob() {
-			return join(this.config.dir, "**/(page|layout).tsx");
-		}
-		toPath(src) {
-			const routePath = cleanPath(src, this.config)
-				// remove the initial slash
-				.replace(/\/page$/, "/")
-				.replace(/\/layout$/, "")
-				.replace(/index$/, "")
-				.replace(/\[([^\/]+)\]/g, (_, m) => {
-					if (m.length > 3 && m.startsWith("...")) {
-						return `*${m.slice(3)}`;
-					}
-					if (m.length > 2 && m.startsWith("[") && m.endsWith("]")) {
-						return `$${m.slice(1, -1)}?`;
-					}
-					return `$${m}`;
-				});
+const resolveToRelative = (p) => {
+	const toAbsolute = (file) => file.split("://").at(-1);
 
-			return routePath?.length > 0 ? `${routePath}` : "/";
-		}
+	const resolved = toAbsolute(resolve(p, import.meta.url));
 
-		toRoute(src) {
-			let path = this.toPath(src);
+	const relative = path.relative(
+		path.resolve(toAbsolute(import.meta.url), ".."),
+		resolved,
+	);
 
-			const [_, exports] = analyzeModule(src);
-			const hasLoader = exports.find((e) => e.n === "loader");
-			const hasErrorBoundary = exports.find((e) => e.n === "ErrorBoundary");
-			const hasLoading = exports.find((e) => e.n === "Loading");
-			const hasConfig = exports.find((e) => e.n === "config");
-			return {
-				$component: {
-					src: src,
-					pick: ["default", "$css"],
-				},
-				$error: hasErrorBoundary
-					? { src: src, pick: ["ErrorBoundary"] }
-					: undefined,
-				$loading: hasLoading ? { src: src, pick: ["Loading"] } : undefined,
-				$$loader: hasLoader
-					? {
-							src: src,
-							pick: ["loader"],
-					  }
-					: undefined,
-				$$config: hasConfig ? { src: src, pick: ["config"] } : undefined,
-				path,
-				filePath: src,
-			};
-		}
-	}
-
-	return (router, app) =>
-		new TanstackFileRouter(
-			{
-				dir: resolve.absolute(config.dir, router.root),
-				extensions: config.extensions ?? ["ts", "tsx", "jsx", "js"],
-			},
-			router,
-			app,
-		);
-}
+	return relative;
+};
 
 export default createApp({
-	server: {},
 	routers: [
 		{
 			name: "public",
-			type: "static",
```

Comment:
i don't think this is correct, the new API is "type", "mode" is old, i think you will have to bump to vinxi": "0.3.9", and make it work with that

Reply:
Done, thanks. Should the validation schema not throw if it is being passed a wrong option tho? Or if it is for backwards compatibility reasons to at least print a deprecation warning.

Reply:
Yeah I think makes sense to do a deprecation warning, I just don't want to add those while we are still in 0.x.x versions, coz things are changing and can't add notices for everything, the validation should error (but I think for now we accepted both until I remove one) okay yeah maybe then a deprecation warning is a good idea, I think i will remove the option and just error in the validation)

Reply:
Great, yeah erroring sounds good to me